test/integration/targets/yum/tasks/repo.yml

```diff
@@ -588,3 +588,58 @@
         state: absent
 
   when: ansible_pkg_mgr == 'yum'
+
+# https://github.com/ansible/ansible/issues/45250
+- block:
+    - name: Install foo-1.0, foo-bar-1.0, bar-1.0
+      yum:
+        name: "foo-1.0,foo-bar-1.0,bar-1.0"
+        state: present
+
+    - name: Upgrade foo*
+      yum:
+        name: foo*
+        state: latest
+      register: yum_result
+
+    - name: Check foo with rpm
+      shell: rpm -q foo
+      register: rpm_result
+
+    - name: Verify update of foo
+      assert:
+        that:
+            - "rpm_result.stdout.startswith('foo-1.1-1')"
+
+    - name: Check foo-bar with rpm
+      shell: rpm -q foo-bar
+      register: rpm_result
+
+    - name: Verify update of foo-bar
+      assert:
+        that:
+            - "rpm_result.stdout.startswith('foo-bar-1.1-1')"
+
+    - name: Check bar with rpm
+      shell: rpm -q bar
+      register: rpm_result
+
+    - name: Verify bar did NOT get updated
+      assert:
+        that:
+            - "rpm_result.stdout.startswith('bar-1.0-1')"
+
+    - name: Verify yum module outputs
+      assert:
+        that:
+            - "yum_result is changed"
+            - "'msg' in yum_result"
+            - "'rc' in yum_result"
+            - "'results' in yum_result"
+  always:
+    - name: Clean up
+      yum:
+        name: foo,foo-bar,bar
+        state: absent
+
+  when: ansible_pkg_mgr == 'yum'
```

Comment:
Any specific reason to not allow this to run for the `dnf` backend also?

Reply:
No, just copy&paste without realizing that. Do you want me to create a follow up PR for that?

Reply:
@mkrizek yes please :)

Reply:
https://github.com/ansible/ansible/pull/48607